File: BTP-java-spring/src/main/java/btp/model/MaitreOeuvre.java
package btp.model;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

@Entity
@DiscriminatorValue("MOeuvre")
public class MaitreOeuvre extends Societe {
	
	@OneToMany(mappedBy = "maitreOeuvre")
	private List<Offre> offres = new ArrayList<Offre>();

	@OneToMany(mappedBy = "maitreOeuvre")
	private List<Facture> facture = new ArrayList<Facture>();

	// Constructeur
	
	public MaitreOeuvre() {
		super();
	}
	
	public MaitreOeuvre(String nom, String siret, String telephone, String mail, String numeroDeTva) {
		super(nom, siret, telephone, mail, numeroDeTva);
	}

	// Getters and Setters

	public List<Offre> getOffres() {
		return offres;
	}

	public void setOffres(List<Offre> offres) {
		this.offres = offres;
	}

	public List<Facture> getFacture() {
		return facture;
	}

	public void setFacture(List<Facture> facture) {
		this.facture = facture;
	}
}

File: BTP/src/app/model/maitreOeuvre.ts
import {Societe} from "./societe";
import {Offre} from "./offre";
import {Facture} from "./facture";
import {Utilisateur} from "./utilisateur";
import {Adresse} from "./adresse";

export class MaitreOeuvre extends Societe{
  // offres: Array<Offre>;
  // factures: Array<Facture>;


  constructor(id?: number, version?: number, nom?: string, siret?: string, telephone?: string, mail?: string, numeroDeTva?: string, utilisateur?: Utilisateur, adresse?: Adresse, type?: string) {
    super(id, version, nom, siret, telephone, mail, numeroDeTva, utilisateur, adresse, type);
  }
}
